src/components/Data/Collections/CreateOrUpdate.vue

```diff
@@ -1,137 +1,232 @@
-          <div class="card error red-color white-text">
-            <i class="fa fa-times dismiss-error" @click="dismissError()" />
-            An error occurred while
-            {{ $route.params.collection ? 'updating' : 'creating' }} collection:
-            <br />{{ error ? error : mappingError }}
-          </div>
-        </div>
+      <div v-show="!realtimeOnlyState">
+        <b-form-group label="Dynamic mapping" label-cols-sm="3">
+          <template v-slot:description
+            >Set the type of dynamic policy for this collection.
+            <a
+              target="_blank"
+              href="https://docs.kuzzle.io/core/2/guides/essentials/database-mappings/#dynamic-mapping-policy"
+              >Read more about Dynamic Mappings</a
+            >.
+          </template>
+          <b-form-radio-group
+            class="pt-2"
+            v-model="dynamicState"
+            :options="['true', 'false', 'strict']"
+          ></b-form-radio-group>
+        </b-form-group>
+        <hr />
+        <b-row>
+          <b-col cols="8">
+            <json-editor
+              id="collection"
+              ref="jsoneditor"
+              tabindex="4"
+              myclass="pre_ace"
+              :content="mappingState"
+            />
+          </b-col>
+
+          <b-col cols="4">
+            <div class="text-secondary">
+              You can (optionally) use this editor to define the mapping for
+              this collection.
+              <br />
+              The mapping of a collection is the definition of how each document
+              in the collection (and its fields) are stored and indexed.
+              <a
+                href="https://docs.kuzzle.io/api/1/controller-collection/update-mapping/"
+                target="_blank"
+                >Read more about mapping</a
+              >
+              <br /><br />
+              For example:
+              <pre>
+{
+"age": { "type": "integer" },
+"name": { "type": "text" }
+}
+        </pre
+              >
+            </div>
+          </b-col>
+        </b-row>
       </div>
-    </div>
+    </b-card>
   </div>
 </template>
 
 <script>
 import Headline from '../../Materialize/Headline'
 import Focus from '../../../directives/focus.directive'
-import Stepper from '../../Common/Stepper'
-import Mapping from './Steps/Mapping'
-import CollectionForm from './Steps/CollectionForm'
+import JsonEditor from '../../Common/JsonEditor'
 
 export default {
   name: 'CollectionCreateOrUpdate',
   components: {
     Headline,
-    Stepper,
-    Mapping,
-    CollectionForm
+    JsonEditor
   },
   directives: {
     Focus
   },
   props: {
-    error: String,
-    index: String,
-    headline: String
+    index: { type: String, required: true },
+    collection: String,
+    headline: String,
+    submitLabel: { type: String, default: 'OK' },
+    realtimeOnly: Boolean,
+    mapping: {
+      type: Object,
+      default: () => ({})
```

Comment:
I've nothing to say about that but I'm interested about why we could not use `default: {}`

Reply:
Actually, Vuejs forces the dev to do it this way.